### ISSUE TITLE
macOS 13.0 Ventura update means PS5 DualSense Controller won’t configure

### ISSUE THREAD

**Issue by j-hussain** [OpenEmu/OpenEmu#4790]
Prior to the update, I’d connect my DualSense controller via Bluetooth to play the games but after updating my laptop, OpenEmu won’t register any inputs. The Bluetooth connection is stable, and OpenEmu detects the controller in the `Controls` panel, but none of the buttons seem to work. The issue isn’t with the controller as it works on other apps on my MacBook. This is on OpenEmu v2.3.3.

Any way to fix this?

**Comment by steffen-klein** [comment]
Same for me with DualShock 4 controller in Ventura and Bluetooth. DIrect USB connection works

**Comment by gro322** [comment]
Same here, please help with a fix when someone found it or will wait until a OE update luckily

**Comment by BIshounen** [comment]
This is not a fix but Erasing Controller in Game Controller did the trick for me:
1. Disconnect controller
2. System Settings -> Game Controller -> Erase
3. Connect controller. From this point it will work both in OpenEmu and RetroArch

I have Dual Sense, connected via bluetooth (usb connected worked before)
Hope this helps.

**Comment by Japandave** [comment]
Same for me, but with the 8bitdo. After upgrading to 13.0.1, Openemu stopped accepting input from my controller. Steam recognizes it fine, so I know it still works, but Openemu no longer does. 

And I did the Erase Controller Setting that Bishounen suggests, but it didn't fix the issue for me.

**Comment by BIshounen** [comment]
> And I did the Erase Controller Setting that Bishounen suggests, but it didn't fix the issue for me.

Have you closed Steam? Every time I run Steam with controller connected (or connect controller when steam is opened) OpenEmu stops recognising it and I need to erase settings again. Speaking of dual sense if its "grabbed" by Steam it shines cyan, while when it's not used by Steam it is blue or whatever color is set up in OS settings.

**Comment by Japandave** [comment]
> Have you closed Steam?

Yep. To make sure it was clear, I rebooted too after deleting the controller settings. But no, OpenEmu still won't recognize it. I guess I will just have to use it with the USB cable connected until this bug is resolved.

**Comment by RobinWragg** [comment]
Dualsense doesn't work for me either. I've added OpenEmu to both input monitoring an bluetooth privacy settings.

**Comment by tr1s** [comment]
> This is not a fix but Erasing Controller Settings in Game Controller did the trick for me:
> 
> 1. Disconnect controller
> 2. System Settings -> Game Controller -> Select Controller -> Erase Controller Settings
> 3. Connect controller. From this point it will work both in OpenEmu and RetroArch
> 
> I have Dual Sense, connected via bluetooth (usb connected worked before) Hope this helps.

**This fix doesn't work for me.** PS5 controller, Ventura 3.4 (22F66). Bluetooth.

Does anyone know what else I can try? Is OpenEmu no longer under development?

**Comment by janlishak** [comment]
The fix worked for me ( MacBook Pro M2 Ventrua 13.3.1) and PS5 Controller
1. System Settings -> Game Controller -> Select Controller -> Erase Controller Settings
2. Bluetooth > forget device
3. Pair again
I'm not sure why this fixed the problem, but since I did this, it works perfectly.

**Comment by DarkWolfTail** [comment]
> The fix worked for me ( MacBook Pro M2 Ventrua 13.3.1) and PS5 Controller
> 
> 1. System Settings -> Game Controller -> Select Controller -> Erase Controller Settings
> 2. Bluetooth > forget device
> 3. Pair again
>    I'm not sure why this fixed the problem, but since I did this, it works perfectly.

I also had problem with my controller, that helped me

**Comment by Davidjordao2019** [comment]
> > This is not a fix but Erasing Controller Settings in Game Controller did the trick for me:
> > 
> > 1. Disconnect controller
> > 2. System Settings -> Game Controller -> Select Controller -> Erase Controller Settings
> > 3. Connect controller. From this point it will work both in OpenEmu and RetroArch
> > 
> > I have Dual Sense, connected via bluetooth (usb connected worked before) Hope this helps.
> 
> THIS - worked for me make the PS4 DualSense Controller work with OpenEmu 👍

worked. Thanks

**Comment by pitis** [comment]
I've opened OpenEmu/OpenEmu-SDK#38 for this.

**Root cause.** The DualSense speaks two different HID dialects. Over USB it sends
input report `0x01`, whose stick, trigger, hat and button fields the report
descriptor describes with ordinary Generic Desktop and Button usages — which is why
wired works. Over Bluetooth it sends a cut-down `0x01` only until something reads one
of its feature reports, then switches to input report `0x31` for the rest of the
connection. Report `0x31`'s payload is declared as vendor-defined data (usage page
`0xFF00`), so IOKit exposes no Generic Desktop or Button elements for it at all, and
macOS appears to trigger that switch on connect.

Here is the relevant part of the descriptor read off a real DualSense over Bluetooth:

```
8501      Report ID (1)
0930 0931 0932 0935   X, Y, Z, Rz     ← the only Generic Desktop axes
0939                  Hat switch       ← the only hat
0509 1901 290e        Buttons 1-14     ← the only buttons
0600ff  Usage Page (Vendor Defined FF00)
954d    Report Count (77)
8531      Report ID (0x31)
093b 8102 INPUT                        ← 77 vendor-defined bytes, no usages
```

Every element OpenEmu resolves its controls through belongs to report `0x01` — the
report the controller has stopped sending. The elements never change value, so no
event is ever dispatched. That is exactly the reported symptom: the controller
enumerates, shows a full control list in Preferences → Controls, and does nothing.

**The fix** decodes the raw input reports instead, the way
`OESwitchProControllerHIDDeviceHandler` already does for the Switch Pro Controller.
Only Bluetooth is claimed, so the working USB path is untouched.

**Verified** on a DualSense (VID `0x054C`, PID `0x0CE6`) on macOS 26.6, which sends
report `0x31` exclusively — 383 reports in 6 seconds, not one `0x01`. After the
change, 381 reports per 4 seconds reach the handler and decode correctly, and 13
distinct controls were confirmed dispatching as their correctly named bindings.
